utils/frame/generate-bags/src/lib.rs

```diff
@@ -0,0 +1,227 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Support code to ease the process of generating bag thresholds.
+//!
+//! NOTE: this assume the runtime implements [`pallet_staking::Config`], as it requires an
+//! implementation of the traits [`frame_support::traits::Currency`] and
+//! [`frame_support::traits::CurrencyToVote`].
+//!
+//! The process of adding bags to a runtime requires only four steps.
+//!
+//! 1. Update the runtime definition.
+//!
+//!    ```ignore
+//!    parameter_types!{
+//!         pub const BagThresholds: &'static [u64] = &[];
+//!    }
+//!
+//!    impl pallet_bags_list::Config for Runtime {
+//!         // <snip>
+//!         type BagThresholds = BagThresholds;
+//!    }
+//!    ```
+//!
+//! 2. Write a little program to generate the definitions. This program exists only to hook together
+//! the runtime definitions with the various calculations here. Take a look at
+//! _utils/frame/node-runtime_ for an example.
```

Comment:
I can't find this file
```suggestion
//! _utils/frame/generate_bags/node-runtime_ for an example.
```